Lib/fontTools/varLib/__init__.py

```diff
@@ -315,13 +319,23 @@ def _add_gvar(font, masterModel, master_ttfs, tolerance=0.5, optimize=True):
 		endPts = control[1] if control[0] >= 1 else list(range(len(control[1])))
 
 		for i,(delta,support) in enumerate(zip(deltas[1:], supports[1:])):
-			if all(abs(v) <= tolerance for v in delta.array):
+			if all(abs(v) <= tolerance for v in delta.array) and not isComposite:
 				continue
 			var = TupleVariation(support, delta)
 			if optimize:
 				delta_opt = iup_delta_optimize(delta, origCoords, endPts, tolerance=tolerance)
 
 				if None in delta_opt:
+					"""In composite glyphs, there should be one 0 entry
+					to make sure the gvar entry is written to the font.
+
+					This is to work around an issue with macOS 10.14 and can be
+					removed once the behaviour of macOS is changed.
+
+					https://github.com/fonttools/fonttools/issues/1381
+					"""
+					if all(d is None for d in delta_opt):
+						delta_opt = [(0, 0)] + [None] * (len(delta_opt) - 1)
```

Comment:
@behdad @jenskutilek 
I was rereading this today, and I think that it doesn't make sense that here we try to run `iup_delta_optimize` on a composite glyph's deltas. The gvar spec is clear in saying that "inferring of deltas for un-referenced points applies only to simple glyphs, not to composite glyphs"
varLib should just write the composite glyph deltas as they are, without any inferred delta. Well ok, if they are all (0, 0) we can write the first (0, 0) and the rest as None, but that's about it.
iup_delta_optimize should only be invoked on simple glyph deltas.

Reply:
@behdad in an earlier comment wrote, referring to the gvar spec sentence I quoted above

> What that means simply is that if a point has no deltas in gvar for composite glyphs, then the value should be assumed to be zero.

I actually couldn't find any reference in the gvar spec saying that when a point has no delta for composite glyphs it's assumed to be (0, 0). I wish this could be clarified in the spec.

If that's confirmed, then we need two different optimization methods for gvar deltas: one `iup_delta_optimize` that is to be used for simple contour glyphs, and another for composite glyphs that replaces (0,0) deltas by `None`; and if all None, leave at least the first delta as (0, 0) to keep the gvar entry for the composite glyph and make macOS happy.

Reply:
> I was rereading this today, and I think that it doesn't make sense that here we try to run `iup_delta_optimize` on a composite glyph's deltas. The gvar spec is clear in saying that "inferring of deltas for un-referenced points applies only to simple glyphs, not to composite glyphs"

I didn't notice these comments before...

We already handle what you specify in glyf.getCoordinatesAndControl. Basically we assign position of each component its own contour, by way of this line:
```
                if glyph.isComposite():
                        coords = GlyphCoordinates(
                                [(getattr(c, 'x', 0), getattr(c, 'y', 0)) for c in glyph.components]
                        )
                        controls = _GlyphControls(
                                numberOfContours=glyph.numberOfContours,
                                endPts=list(range(len(glyph.components))),
                                flags=None,
                                components=[c.glyphName for c in glyph.components],
                        )
```

A single-point contour never gets any interpolation done for it. So that part is taken care of. We still can call iup optimization on it because a contour that is not moved at all, can avoid encoding the 0,0 movement, at the expense of having to encode point numbers explicitly. So it still can possibly benefit from IUP.

The bug seems to have simply been a wrong optimization in MacOS. That's all.